### PR TITLE
some suggestions for refactoring

### PR DESCRIPTION
**Pull request by bbonf**

### CODE REVIEW

File: backend/spindle/views.py [hunk 1]
```diff
@@ -38,92 +43,102 @@ def json_response(self) -> JsonResponse:
...
+            logging.warn("Input is not JSON parseable: %s", request_body)
+            return
+
+        if not "input" in parsed_json or not isinstance(parsed_json["input"], str):
+            logging.warn("Key input with value of type string not found:", request_body)
+            return
+
+        return parsed_json['input']
+
+    def latex_response(self, tex):
```
Comment (XanderVertegaal) [inline backend/spindle/views.py:113]: You may want to add return types to this function signature (and the ones following this one).

**Review by XanderVertegaal — APPROVED** [review]
Good suggestions; very neat! :) Feel free to merge!

I have a few small, nitpicky suggestions, but you should feel free to ignore these.

File: backend/spindle/views.py [hunk 2]
```diff
@@ -38,92 +43,102 @@ def json_response(self) -> JsonResponse:
...
-        return SpindleResponse(error=SpindleErrorSource.GENERAL).json_response()
+    def read_request(self, request) -> Optional[str]:
+        """Read and validate the HTTP request received from the frontend"""
+        request_body = request.body.decode("utf-8")
+
+        try:
+            parsed_json = json.loads(request_body)
+        except json.JSONDecodeError:
+            logging.warn("Input is not JSON parseable: %s", request_body)
+            return
```
[the remark this answers is not in the document]
Comment (jgonggrijp) [reply backend/spindle/views.py]: Huh? I thought `return` returned `None` by default?

[the remark this answers is not in the document]
Comment (XanderVertegaal) [reply backend/spindle/views.py]: It does, so the code itself is fine and does not break. It is more of a style thing. I have been taught that simple returns are for 'early out's, when you don't really care about what comes out of the function. With `return None`, you suggest that you are going to check the return value somewhere.

But maybe I am being too pedantic here.

**Pull request merged**